### DISCUSSION TITLE
How to correctly pass mold options

### DISCUSSION THREAD

**Discussion by ghost** [rui314/mold#1327] [General]
Hi. I've been playing with mold on a Arch Linux box for a while now. Works fantastic. I noticed `2.33.0` got a new  [--separate-debug-file](https://github.com/rui314/mold/releases/tag/v2.33.0) option that could bring a nice speed-gain and I wanted to test that.

The release notes mentions `--separate-debug-info`, and that confused me at first. Running `mold --help` indicates this option is actually called `--separate-debug-file`, as can be seen [here](https://github.com/rui314/mold/blob/f18d1c1ffcee937a8ba735c4c594f2f9c308c319/elf/cmdline.cc#L150).

On the [Arch wiki page for mold](https://wiki.archlinux.org/title/Makepkg#Using_mold_linker) it is advised to append `-fuse-ld=mold` to LDFLAGS in /etc/makepkg.conf. Which is how I've been running mold for a while now.

My question: what would be the correct/proper way to pass the `--separate-debug-file` option in such a setup?

TIA

**Comment by rui314** [comment]
I fixed the typo in the release notes. Thank you for letting me know!

You are not supposed to directly invoke the linker. Usually it is invoked through the compiler frontend (`cc`, `gcc` or `clang`). Try to add `-Wl,--separate-debug-file` to your command option. The compiler frontend recognizes `-Wl,` as a linker option and pass the string that follows it to the linker.

**Reply by ghost** [reply]
Excellent. Passing `-Wl,--separate-debug-file` in Arch's makepkg.conf did the trick.

Thank you for the speedy reply and the wonderful work on mold.

**Comment by rgwood** [comment]
In case anyone's looking to enable this for Rust, I _think_ adding `"-C", "link-arg=-Wl,--separate-debug-file"` to config.toml is the way to do it. For example:

```
[target.x86_64-unknown-linux-gnu]
linker = "clang"
rustflags = ["-C", "link-arg=-fuse-ld=/home/linuxbrew/.linuxbrew/bin/mold", "-C", "link-arg=-Wl,--separate-debug-file"]

```
